interactions/client/get.py

```diff
@@ -0,0 +1,397 @@
+
+            elif force_http:
+                _objects.clear()
+                _func = getattr(http_name, client._http)
+                for _id in kwargs.get(kwarg_name):
+                    _kwargs = kwargs
+                    _kwargs.pop(kwarg_name)
+                    _kwargs[kwarg_name[:-1]] = _id
+                    _objects.append(_func(**_kwargs))
+                return _http_request(_obj, http=client._http, request=_objects)
+
+            else:
+                _func = getattr(http_name, client._http)
+                for _index, __obj in enumerate(_objects):
+                    if __obj is None:
+                        _id = kwargs.get(kwarg_name)[_index]
+                        _kwargs = kwargs
+                        _kwargs.pop(kwarg_name)
+                        _kwargs[kwarg_name[:-1]] = _id
+                        _request = _func(**_kwargs)
+                        _objects[_index] = _request
+                return _http_request(_obj, http=client._http, request=_objects)
+
+        _obj: Optional[_T] = None
+
+        force_cache = kwargs.pop("force", None) == "cache"
+        force_http = kwargs.pop("force", None) == "http"
+        if not force_http:
+            _obj = _get_cache(obj, client, kwarg_name, **kwargs)
+
+        if force_cache:
+            return _obj
+
+        elif not force_http and _obj:
+            return _return_cache(_obj)
+
+        else:
+            return _http_request(obj=obj, http=client._http, _name=http_name, **kwargs)
+
+    elif len(args) == 1:
+        return _search_iterable(*args, **kwargs)
+
+
+async def _http_request(
+    obj: Type[_T],
+    http: HTTPClient,
+    request: Union[Coroutine, List[Union[_T, Coroutine]], List[Coroutine]] = None,
+    _name: str = None,
+    **kwargs,
+) -> Union[_T, List[_T]]:
+    if not request:
+        if obj in (Role, Emoji):
+            _guild = Guild(**await http.get_guild(kwargs.pop("guild_id")), _client=http)
+            _func = getattr(_guild, _name)
+            return await _func(**kwargs)
+
+        _func = getattr(http, _name)
+        _obj = await _func
+        return obj(**_obj, _client=http)
+
+    if not isinstance(request, list):
+        return obj(**await request, _client=http)
+
+    return [obj(**await req, _client=http) if isawaitable(req) else req for req in request]
+
+
+async def _return_cache(
+    obj: Union[Optional[_T], List[Optional[_T]]]
+) -> Union[Optional[_T], List[Optional[_T]]]:
+    await sleep(0)  # iirc Bluenix meant that any coroutine should await at least once
+    return obj
+
+
+def _get_cache(
+    _object: Type[_T], client: Client, kwarg_name: str, _list: bool = False, **kwargs
+) -> Union[Optional[_T], List[Optional[_T]]]:
+    if _list:
+        _obj = []
+        if _object == Member:  # Can't be more dynamic on this
+            _values = ()
+            _guild_id = Snowflake(kwargs.get("guild_id"))
+            for _id in kwargs.get("member_ids"):
+                _values += (
```

Comment:
Why are you using a tuple to gather the values? Using a list and appending to it is much faster, and lists are made to do things like this

Reply:
Because from my last understanding a member needed a Tuple of guild and member id, no?

Reply:
Yes, but you can have a list of tuples. As long as the key you're using is a tuple of the right ids, it doesn't matter where/how you get/store it